src/advect1d_vertical.jl

```diff
@@ -0,0 +1,253 @@
+#### This Julia file contains functions to implement 
+####  1) 1-D L94 vertical advection: advect1d_l94() (Lin et al., 1994)
+####  2) 1-D PPM vertical advection: advect1d_ppm() (Collela and Woodward, 1984)
+####  The vertical advection conserve pi*q rather than q. (Lin and Rood, 1996)
+
+#### I will upload the environment later but basically the code block below is the packages I use.
+##   -------------------------------------------------------------------------------------------
+##   ]activate EarthSciML_Advection
+##   using EarthSciData, EarthSciMLBase, GasChem,
+##   DomainSets, ModelingToolkit, DifferentialEquations, Dates
+##   using Plots, Statistics, GeoDataFrames, NCDatasets
+##   using Flux
+##   using BSON: @save
+##   using BSON: @load
+##   using Base.Threads
+##   using LoopVectorization
+##   using Tullio
+##   -------------------------------------------------------------------------------------------
+
+
+
+
+## L94 advection in 1-D
+## This is to make splitted scheme
+function advect1d_vertical_L94(s1d_in,vel_1d,nz,Δt,p)
+
+    
+    ϕ1 = zeros(Float64, nz+5)
+    ϕ2 = zeros(Float64, nz+5)
+    δϕ1 = zeros(Float64, nz+5)
+    Δϕ1_avg = zeros(Float64, nz+4)
+    Δϕ1_mono = zeros(Float64, nz+3)
+    δϕ2 = zeros(Float64, nz+4)
+    Δϕ2_avg = zeros(Float64, nz+3)
+    Δϕ2_mono = zeros(Float64, nz+3)
+    FLUX = zeros(Float64, nz+1)
+    U = zeros(Float64, nz+1)
+    courant = zeros(Float64, nz+1)
+    ΔP = zeros(Float64, nz+2)
+    s1d_out = zeros(Float64, nz+5)
+        
+    #U[1] = vel_1d[1]; U[nz+1] = vel_1d[nz];  ## In case we want Neumann BC.
+    U[1] = 0; U[nz+1] = 0;  ## Dirichlet BC with zero on top and bottom of the column.
+    
+    for i in 2:nz
```

Comment:
Add comments before each of these blocks briefly describing what they do, and referencing equation numbers in the paper as appropriate.